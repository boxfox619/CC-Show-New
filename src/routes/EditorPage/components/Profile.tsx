import * as React from 'react';
import styled from 'styled-components';

const ProfileContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: left;
    padding: 10%;
`
const Thumbnail = styled.div`
    border-radius: 50%;
    width: 3em;
    height: 3em;
    display: inline-block;
    overflow: hidden;
    position: relative;
    margin-right: 10px;
    & > img {
        position: absolute;
        width: 3em;
        top: 50%;
        left: 50%;
        transform: translate(-50%, -50%);
    }
`
const Name = styled.div`
    font-size: 1.2em;
    font-weight: bold;
    background: linear-gradient(to right, #fc00ff, #00dbde);
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
`
const SubName = styled.div`
    font-size: 0.5em;
    color: #94a8ba;
`

interface Props {
    name: string,
    subName: string,
    thumbnail: string
}

const Profile: React.FC<Props> = (props: Props) => {
    return (
        <ProfileContainer>
            <Thumbnail><img src={props.thumbnail} alt="thumbnail" /></Thumbnail>
            <div>
                <Name>{props.name}</Name>
                <SubName>{props.subName}</SubName>
            </div>
        </ProfileContainer>
    )
}

export default Profile;