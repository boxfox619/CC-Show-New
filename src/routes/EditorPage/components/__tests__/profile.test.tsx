import * as React from 'react';
import * as Enzyme from 'enzyme';
import Profile from '../Profile';

describe('<Profile />', () => {
  it('should render one <div> & div label', () => {
    const wrapper = Enzyme.shallow(<Profile name={'홍길동'} subName={'치킨맥주'}/>);
    expect(wrapper).toMatchSnapshot();
  });
});