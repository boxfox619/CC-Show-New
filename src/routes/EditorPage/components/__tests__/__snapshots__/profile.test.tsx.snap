// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GradientButtonItem /> should match snapshot 1`] = `ShallowWrapper {}`;
